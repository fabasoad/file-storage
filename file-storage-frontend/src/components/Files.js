import React from 'react';
import File from './File';

const Files = (props) => (
    <div>
        <div className="widget-header">
            <h3 className="widget-header__title">Your files</h3>
            <button
                className="button button--link"
                onClick={props.handleDeleteFiles}
            >
                Remove all
            </button>            
        </div>
        {props.files.length === 0 && <p className="widget__message">Please upload a file to get started</p>}
        {
            props.files.map((file, index) =>
                <File
                    key={file}
                    fileName={file}
                    count={index + 1}
                    handleDeleteFile={props.handleDeleteFile}
                />)
        }
    </div>
);

export default Files;