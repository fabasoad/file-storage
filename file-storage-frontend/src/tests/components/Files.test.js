import React from 'react';
import { shallow } from 'enzyme';
import Files from '../../components/Files';

let wrapperWithFiles, wrapperWithoutFiles, expectedFiles, handleDeleteFiles;

const createFileObject = (fileName, index, handleDeleteFile) => ({
    file: fileName,
    index: index,
    handleDeleteFile: handleDeleteFile
});

beforeEach(() => {
    const handleDeleteFile = jest.fn();
    expectedFiles = [
        createFileObject('file1', 1, handleDeleteFile),
        createFileObject('file2', 2, handleDeleteFile)
    ];
    handleDeleteFiles = jest.fn();
    wrapperWithFiles = shallow(<Files
        files={expectedFiles}
        handleDeleteFiles={handleDeleteFiles}
        handleDeleteFile={handleDeleteFile}
    />);
    wrapperWithoutFiles = shallow(<Files
        files={[]}
        handleDeleteFiles={handleDeleteFiles}
        handleDeleteFile={handleDeleteFile}
    />);
});

test('should render empty Files correctly', () => {
    expect(wrapperWithoutFiles).toMatchSnapshot();
});

test('should render not empty Files correctly', () => {
    expect(wrapperWithFiles).toMatchSnapshot();
});

test('should handle handleDeleteFiles', () => {
    wrapperWithoutFiles.find('button').prop('onClick')();
    expect(handleDeleteFiles).toHaveBeenCalledTimes(1);
});