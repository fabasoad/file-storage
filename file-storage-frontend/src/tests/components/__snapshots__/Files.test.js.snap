// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render empty Files correctly 1`] = `
<div>
  <div
    className="widget-header"
  >
    <h3
      className="widget-header__title"
    >
      Your files
    </h3>
    <button
      className="button button--link"
      onClick={[MockFunction]}
    >
      Remove all
    </button>
  </div>
  <p
    className="widget__message"
  >
    Please upload a file to get started
  </p>
</div>
`;

exports[`should render not empty Files correctly 1`] = `
<div>
  <div
    className="widget-header"
  >
    <h3
      className="widget-header__title"
    >
      Your files
    </h3>
    <button
      className="button button--link"
      onClick={[MockFunction]}
    >
      Remove all
    </button>
  </div>
  <File
    count={1}
    fileName={
      Object {
        "file": "file1",
        "handleDeleteFile": [MockFunction],
        "index": 1,
      }
    }
    handleDeleteFile={[MockFunction]}
    key="[object Object]"
  />
  <File
    count={2}
    fileName={
      Object {
        "file": "file2",
        "handleDeleteFile": [MockFunction],
        "index": 2,
      }
    }
    handleDeleteFile={[MockFunction]}
    key="[object Object]"
  />
</div>
`;
